### DISCUSSION TITLE
Default branch named "master" despite my account preferences

### DISCUSSION THREAD

**Discussion by mutevsky** [community/community#60619] [New to GitHub]
I've confirmed that my default branch is set to `main`:
![image](https://github.com/community/community/assets/86967749/42043105-3178-4b78-b37c-1d1c46d753ea)

However, when I create a new repo (via the New Repository button in the top right, on GitHub) and clone it, I've discovered that the default branch is still named `master` anyway!

git status returns:
![image](https://github.com/community/community/assets/86967749/d8953283-8c66-40b6-aa8a-dfb71f6771ce)

`git push origin "main"` doesn't work to push changes; I need to use `master`.

I've renamed it using the sequence:
`git branch -m master main`
`git fetch origin`
`git branch -u origin/main main`
`git remote set-head origin -a`

after renaming it from GitHub, but... _why is this happening_ when I've set up GitHub to name the default branch `main`? As is probably apparent, I'm pretty new to git, and I'm sure I'm missing something dumb here, but if there's any way to avoid this annoyance in the future, I'd love to find out. Thanks!

**Comment by Beargruug** [comment]
This issue occurs because when you create a new repository and clone it, Git uses the default branch name specified by its own configuration, not the default branch name set in GitHub repository settings. To avoid this annoyance in the future, you can follow these steps:

```
git branch -m master main
git push -u origin main
``` 
The first command renames the branch locally, and the second command pushes the renamed branch to the remote repository on GitHub.

By following these steps, you will ensure that the default branch in both your local repository and the remote repository on GitHub is named "main."

**Comment by ramamimu** [comment]
old version of github has default branch name `master`. But today it is changes to `main`.
If you have a problem in your locally git. It happen because difference branch name.

You probably have two options :
1. change your branch name in every single initiation. But don't worry about the command. It always appear everytime you create new repository on github. you just simply copy to your terminal.
![image](https://github.com/community/community/assets/81814981/8ca4efdb-82a0-4a89-a784-d6d203cd3eec)

The command `git branch -M main` will change branch name into `main` to your current local repository.

2. Change default name on your `git config` with copy this command.
```
git config --global init.defaultBranch main
```
It is automatically change your default name to main instead of master when create new git initialization.

**Reply by dotnetappdev** [reply]
With reasoning behind this change in the first place, cause of slavery you think they would now give you the ability to set a custom new repo default name
